### PR TITLE
Basic support for TLS

### PR DESCRIPTION
**Pull request by datafx**
Added basic support for TLS to the MQTT server and added an example ini file.  I'm not very good with python and I wasn't able to figure out a way to set the TLS protocol in the ini, but it does work.

### CODE REVIEW

**Comment by sumnerboy12** [comment]
Hi, firstly thanks for the PR! Appreciate you taking the time to add this functionality. It looks good, but I would prefer we didn't have two example ini's, just one with a commented section for the new TLS stuff.

As for the TLS version, have a look at this code and perhaps use that as an example of how to support TLS version in the INI file - https://github.com/jpmens/mqttwarn/blob/611c0967b41f7ca10672fb40a40576d145fded7e/mqttwarn.py#L109.

**Comment by datafx** [comment]
Thanks for the info. I will take a look later this week and try to make those changes.

**Comment by datafx** [comment]
Ok, I made some more changes that allow you to set the TLS version from the INI.  Setting MQTT_TLS_PROTOCOL to none should disable TLS and allow you to use a regular unencrypted connection.  I'm not sure if you would prefer a note in the example INI or if you would rather mention it in the readme.

File: mqtt-gpio-monitor.py [hunk 1]
```diff
@@ -238,6 +241,21 @@ def connect():
     if MQTT_USERNAME:
         mqttc.username_pw_set(MQTT_USERNAME, MQTT_PASSWORD)
 
+    # Set TLS details
+    if MQTT_TLS_PROTOCOL is not None:
```
Comment (sumnerboy12) [inline mqtt-gpio-monitor.py]: Simpler to just say `if MQTT_TLS_PROTOCOL:` like I do above for username.

File: mqtt-gpio-monitor.ini.example [hunk 2]
```diff
@@ -3,9 +3,12 @@ MODULE             = <gpio or pfio>
 DEBUG              = True
 
 MQTT_HOST          = localhost
-MQTT_PORT          = 1883
+MQTT_PORT          = 8883
 MQTT_USERNAME      = 
 MQTT_PASSWORD      = 
+MQTT_TLS_PROTOCOL  = tlsv1_2
```
Comment (sumnerboy12) [inline mqtt-gpio-monitor.ini.example]: I think leave these blank by default (and leave port as `1883`) and have a commented section with your TLS defaults

**Comment by datafx** [comment]
Awesome! Thanks for pointing me in the right direction!

**Comment by sumnerboy12** [comment]
Merged, thanks very much for your contribution!

**Comment by datafx** [comment]
No problem.  Glad I could help! :)

**Pull request merged**